packages/common/src/ui/components/popover/BasePopover/usePopover.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { Dispatch, FC, useState, useRef, MutableRefObject } from 'react';
+import { useDebounceCallback } from '../../../../hooks/useDebounce';
+import { BasePopover } from './BasePopover';
+
+type VirtualElement = { getBoundingClientRect: () => DOMRect };
+
+type AnchorElementSetter = React.Dispatch<
+  React.SetStateAction<VirtualElement | null>
+>;
+
+type AnchorElementSetterRef = MutableRefObject<AnchorElementSetter | null>;
+type IsOpenSetterRef = MutableRefObject<Dispatch<
+  React.SetStateAction<boolean>
+> | null>;
+
+interface UsePopoverControl {
+  show: React.MouseEventHandler<HTMLDivElement | HTMLButtonElement>;
+  hide: () => void;
+  Popover: FC;
+}
+
+export const usePopover = (): UsePopoverControl => {
+  // The Popover component itself carries the state and
+  // assigns callbacks to these refs which can control
+  // the state. This is done so that we can control the
+  // state, but also ensure the Popover component doesn't
```

Comment:
:nit.tiny
is this right?
```suggestion
  // state, but also ensures the Popover component doesn't
```